packages/compass-explain-plan/src/components/explain-tree/explain-tree-stage.tsx

```diff
@@ -145,37 +228,42 @@ const ExecutionStats: React.FunctionComponent<ExecutionstatsProps> = ({
 const StageView: React.FunctionComponent<StageViewProps> = (props) => {
   return (
     <>
-      <Subtitle>{props.name}</Subtitle>
-      <HorizontalRule className={separatorStyles} />
+      <div className={stageTitleStyles}>
+        <Icon glyph={props.detailsOpen ? 'ChevronDown' : 'ChevronRight'} />
+        <Subtitle>{props.name}</Subtitle>
+      </div>
 
       <ExecutionStats
         nReturned={props.nReturned}
         prevStageExecTimeMS={props.prevStageExecTimeMS}
         curStageExecTimeMS={props.curStageExecTimeMS}
         totalExecTimeMS={props.totalExecTimeMS}
-      ></ExecutionStats>
+      />
 
       {Object.entries(props.highlights).length > 0 && (
         <div>
           <HorizontalRule className={separatorStyles} />
           <Highlights highlights={props.highlights}></Highlights>
         </div>
       )}
-      <HorizontalRule className={separatorStyles} />
-      <Button
-        type="button"
-        size="xsmall"
-        variant="default"
-        onClick={() => props.onToggleDetailsClick()}
-      >
-        Details
-      </Button>
+
       {props.detailsOpen && (
-        <div className={codeContainerStyles}>
-          <Code copyable={false} language="json" className={codeStyles}>
-            {JSON.stringify(props.details, null, ' ') || '{}'}
-          </Code>
-        </div>
+        <KeylineCard
+          className={codeContainerStyles}
+          onClick={(e) => e.stopPropagation()}
+        >
+          <CodemirrorMultilineEditor
```

Comment:
If you want to switch to multiline editor, you should probably also set minLines to 1, otherwise when collapsing the doc, it still takes a bunch of space on the screen:

![image](https://github.com/mongodb-js/compass/assets/5036933/59c293ee-dc71-49d7-9cfe-77c257466e6e)